src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast instanceof Herbivore && !beast.hidden) {
+      beast.health -= 50;
+    }
+
+    if (beast.health <= 0) {
+      Animal.alive = Animal.alive.filter(animal => animal.name !== beast.name);
```

Comment:
Let's move this logic to method in instance of `Animal` class. 
You can simply clear the array in this filter from animals that have health <= 0

```suggestion
      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```